File: LocaleConnectBackend/meetup/src/main/java/com/localeconnect/app/meetup/model/Meetup.java
package com.localeconnect.app.meetup.model;


import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "meetup")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Meetup {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "creator_id")
    private Long creatorId;
    private String name;
    private String description;
    private Date date;
    @Column(name = "start_time")
    private String startTime;
    @Column(name = "end_time")
    private String endTime;
    private double cost;
    private String location;

    @ElementCollection
    private List<String> spokenLanguages= new ArrayList<>();

    @ElementCollection
    private List<Long> meetupAttendees = new ArrayList<>();
}

File: LocaleConnectFrontend/src/app/model/meetup.ts
export interface Meetup {
  id: number;
  creatorName: string;
  name: string;
  description: string;
  date: Date;
  startTime: string;
  endTime: string;
  cost: number;
  location: string;
  spokenLanguages: string[];
  meetupAttendees: number[];
  rating: number;
  ratingSubmitted?: boolean;
  totalRatings?: number;
  averageRating?: number;
  expand?: boolean;
  isAttending?: boolean;
}
